src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -101,6 +106,91 @@ public function publish(string $body, array $headers = array()): void
         $this->exchange()->publish($body, null, AMQP_NOPARAM, array('headers' => $headers));
     }
 
+    /**
+     * @throws \AMQPException
+     */
+    public function publishForRetry(\AMQPEnvelope $message): bool
+    {
+        if (!isset($this->connectionCredentials['retry'])) {
+            return false;
+        }
+
+        $retryConfiguration = $this->connectionCredentials['retry'];
+        $attemptNumber = ((int) $message->getHeader('symfony-messenger-attempts') ?: 0) + 1;
+
+        if ($this->shouldSetup()) {
+            $this->setupRetry($retryConfiguration, $attemptNumber);
+        }
+
+        $maximumAttempts = $retryConfiguration['attempts'] ?? 3;
+        $routingKey = str_replace('%attempt%', $attemptNumber, $retryConfiguration['routing_key_pattern'] ?? 'attempt_%attempt%');
```

Comment:
Could we normalize all the options values before using it, in the constructor for instance?

Reply:
Yep, it's a good idea.